Unable to compile rustc_binaryen on Windows 10 with x86_64-pc-windows-msvc toolchain

The compilation of rust fails in `rustc_binaryen` when compiling on Windows 10 with x86_64-pc-windows-msvc toolchain (Visual Studio 2017). I tried compiling from a fresh git clone but got the same error.

I was able to compile rust  using a MinGW64 toolchain.

Compilation log with RUST_BACKTRACE=1:
<details>

```
Building stage0 compiler artifacts (x86_64-pc-windows-msvc -> x86_64-pc-windows-msvc)
   Compiling fmt_macros v0.0.0 (file:///C:/Users/zilbuz/projets/rust/src/libfmt_macros)
   Compiling rustc-demangle v0.1.5
   Compiling cc v1.0.3
   Compiling bitflags v1.0.0
   Compiling serialize v0.0.0 (file:///C:/Users/zilbuz/projets/rust/src/libserialize)
   Compiling cfg-if v0.1.2
   Compiling log v0.3.8
   Compiling graphviz v0.0.0 (file:///C:/Users/zilbuz/projets/rust/src/libgraphviz)
   Compiling ar v0.3.1
   Compiling rustc-serialize v0.3.24
   Compiling rustc_driver v0.0.0 (file:///C:/Users/zilbuz/projets/rust/src/librustc_driver)
   Compiling rustc_trans v0.0.0 (file:///C:/Users/zilbuz/projets/rust/src/librustc_trans)
   Compiling stable_deref_trait v1.0.0
   Compiling libc v0.2.33
   Compiling arena v0.0.0 (file:///C:/Users/zilbuz/projets/rust/src/libarena)
   Compiling rustc_back v0.0.0 (file:///C:/Users/zilbuz/projets/rust/src/librustc_back)
   Compiling rustc_incremental v0.0.0 (file:///C:/Users/zilbuz/projets/rust/src/librustc_incremental)
   Compiling unicode-width v0.1.4
   Compiling rustc_platform_intrinsics v0.0.0 (file:///C:/Users/zilbuz/projets/rust/src/librustc_platform_intrinsics)
   Compiling rustc v0.0.0 (file:///C:/Users/zilbuz/projets/rust/src/librustc)
   Compiling syntax v0.0.0 (file:///C:/Users/zilbuz/projets/rust/src/libsyntax)
   Compiling rustc_metadata v0.0.0 (file:///C:/Users/zilbuz/projets/rust/src/librustc_metadata)
   Compiling rustc_cratesio_shim v0.0.0 (file:///C:/Users/zilbuz/projets/rust/src/librustc_cratesio_shim)
   Compiling filetime v0.1.14
   Compiling env_logger v0.4.3
   Compiling rustc_data_structures v0.0.0 (file:///C:/Users/zilbuz/projets/rust/src/librustc_data_structures)
   Compiling owning_ref v0.3.3
   Compiling num_cpus v1.7.0
   Compiling rand v0.3.18
   Compiling cmake v0.1.26
   Compiling miniz-sys v0.1.10
   Compiling build_helper v0.1.0 (file:///C:/Users/zilbuz/projets/rust/src/build_helper)
   Compiling rustc_apfloat v0.0.0 (file:///C:/Users/zilbuz/projets/rust/src/librustc_apfloat)
   Compiling syntax_pos v0.0.0 (file:///C:/Users/zilbuz/projets/rust/src/libsyntax_pos)
   Compiling rls-span v0.4.0
   Compiling rustc_llvm v0.0.0 (file:///C:/Users/zilbuz/projets/rust/src/librustc_llvm)
   Compiling jobserver v0.1.8
   Compiling rustc_binaryen v0.0.0 (file:///C:/Users/zilbuz/projets/rust/src/librustc_binaryen)
   Compiling rls-data v0.13.0
   Compiling flate2 v0.2.20
   Compiling rustc_errors v0.0.0 (file:///C:/Users/zilbuz/projets/rust/src/librustc_errors)
   Compiling proc_macro v0.0.0 (file:///C:/Users/zilbuz/projets/rust/src/libproc_macro)
   Compiling rustc_const_math v0.0.0 (file:///C:/Users/zilbuz/projets/rust/src/librustc_const_math)
   Compiling syntax_ext v0.0.0 (file:///C:/Users/zilbuz/projets/rust/src/libsyntax_ext)
error: failed to run custom build command for `rustc_binaryen v0.0.0 (file:///C:/Users/zilbuz/projets/rust/src/librustc_binaryen)`
process didn't exit successfully: `C:\Users\zilbuz\projets\rust\build\x86_64-pc-windows-msvc\stage0-rustc\release\build\rustc_binaryen-5b38d8ef9a1be3fe\build-script-build` (exit code: 101)
--- stderr
thread 'main' panicked at '
command did not execute successfully, got: exit code: 1

build script failed, must exit now', C:\Users\zilbuz\.cargo\registry\src\github.com-1ecc6299db9ec823\cmake-0.1.26\src\lib.rs:599:4
stack backtrace:
   0: std::sys_common::backtrace::_print
             at C:\projects\rust\src\libstd\sys_common\backtrace.rs:91
   1: std::panicking::default_hook::{{closure}}
             at C:\projects\rust\src\libstd\panicking.rs:383
   2: std::panicking::default_hook
             at C:\projects\rust\src\libstd\panicking.rs:397
   3: std::panicking::rust_panic_with_hook
             at C:\projects\rust\src\libstd\panicking.rs:577
   4: std::panicking::begin_panic<alloc::string::String>
             at C:\projects\rust\src\libstd\panicking.rs:538
   5: std::panicking::begin_panic_fmt
             at C:\projects\rust\src\libstd\panicking.rs:522
   6: cmake::Config::build
   7: cmake::Config::build
   8: cmake::Config::build
   9: POW10
  10: panic_unwind::__rust_maybe_catch_panic
             at C:\projects\rust\src\libpanic_unwind\lib.rs:101
  11: std::rt::lang_start
             at C:\projects\rust\src\libstd\rt.rs:51
  12: __scrt_common_main_seh
             at f:\dd\vctools\crt\vcstartup\src\startup\exe_common.inl:283
  13: BaseThreadInitThunk

warning: build failed, waiting for other jobs to finish...
error: build failed
thread 'main' panicked at 'command did not execute successfully: "C:\\Users\\zilbuz\\projets\\rust\\build\\x86_64-pc-windows-msvc\\stage0/bin\\cargo.exe" "build" "--target" "x86_64-pc-windows-msvc" "-j" "4" "--release" "--features" " llvm" "--manifest-path" "C:\\Users\\zilbuz\\projets\\rust\\src/rustc/Cargo.toml" "--message-format" "json"
expected success, got: exit code: 101', src\bootstrap\compile.rs:882:8
stack backtrace:
   0: std::sys_common::backtrace::_print
             at C:\projects\rust\src\libstd\sys_common\backtrace.rs:91
   1: std::panicking::default_hook::{{closure}}
             at C:\projects\rust\src\libstd\panicking.rs:383
   2: std::panicking::default_hook
             at C:\projects\rust\src\libstd\panicking.rs:397
   3: std::panicking::rust_panic_with_hook
             at C:\projects\rust\src\libstd\panicking.rs:577
   4: std::panicking::begin_panic<alloc::string::String>
             at C:\projects\rust\src\libstd\panicking.rs:538
   5: std::panicking::begin_panic_fmt
             at C:\projects\rust\src\libstd\panicking.rs:522
   6: <bootstrap::compile::Assemble as bootstrap::builder::Step>::run::{{closure}}
   7: <bootstrap::compile::Rustc as bootstrap::builder::Step>::run
   8: bootstrap::builder::Builder::ensure
   9: <bootstrap::compile::Assemble as bootstrap::builder::Step>::run
  10: bootstrap::builder::Builder::ensure
  11: bootstrap::builder::Builder::compiler
  12: <bootstrap::compile::Assemble as bootstrap::builder::Step>::run
  13: bootstrap::builder::Builder::ensure
  14: bootstrap::builder::Builder::compiler
  15: <bootstrap::compile::Std as bootstrap::builder::Step>::make_run
  16: <bootstrap::builder::Builder<'a> as core::ops::deref::Deref>::deref
  17: <bootstrap::builder::Builder<'a> as core::ops::deref::Deref>::deref
  18: bootstrap::builder::Builder::run
  19: bootstrap::Build::build
  20: core::ptr::<impl *mut T>::is_null
  21: panic_unwind::__rust_maybe_catch_panic
             at C:\projects\rust\src\libpanic_unwind\lib.rs:101
  22: std::rt::lang_start
             at C:\projects\rust\src\libstd\rt.rs:51
  23: main
  24: __scrt_common_main_seh
             at f:\dd\vctools\crt\vcstartup\src\startup\exe_common.inl:283
  25: BaseThreadInitThunk
failed to run: C:\Users\zilbuz\projets\rust\build\bootstrap\debug\bootstrap build
Build completed unsuccessfully in 0:07:43
```
</details>
<br>
I don't know what other details I could give so if you need some additionnal informations just ask :)

There isn't any `output` folder.

I tried changing the locale to English, and the compilation still doesn't work, but I have a clearer message this time: `C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\VC\VCTargets\Microsoft.CppBuild.targets(317,5): error MSB3491: Could not write lines to file "emscripten-optimizer.dir\Release\emscript.6EB7F96E.tlog\emscripten-optimizer.lastbuildstate". The specified path, file name, or both are too long. The fully qualified file name must be less than 260 characters, and the directory name must be less than 248 characters.`

I'll try to compile rust from `c:\rust` to see if this works.

For informations, some log files: 
[Rust compilation log](https://github.com/rust-lang/rust/files/1501590/build.log)
[rustc_binaryen/CMakeOutput.log](https://github.com/rust-lang/rust/files/1501594/CMakeOutput.log)

--- a/README.md
+++ b/README.md
@@ -129,6 +129,9 @@ CALL "C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\bin\amd64\vcvars64.
 python x.py build
 ```
 
+If you are seeing build failure when compiling `rustc_binaryen`, make sure the path
+length of the rust folder is not longer than 22 characters.
+
 #### Specifying an ABI
 [specifying-an-abi]: #specifying-an-abi
 

